lib/mongo/monitoring/event/command_failed.rb

```diff
@@ -27,13 +27,23 @@ class CommandFailed < Mongo::Event::Base
         # @return [ String ] command_name The name of the command.
         attr_reader :command_name
 
+        # @return [ BSON::Document ] command The command arguments.
+        attr_reader :command
```

Comment:
For usability - right now I print all 3 types of events and look at it as a human to figure out what is going on. Imagine a system that wanted to log failures - it would have to process all start events, record them somewhere, then for each failure it would need to figure out the matching start event to figure out the command. All of this when the driver has the command trivially available upon operation completion.

Reply:
The request_id and, especially for batched write operations, the operation_id fields serve the purpose of correlating start, succeed, and fail events. The API in the spec was standardized for use by APM tools like New Relic and App Dynamics. We can't change the API in one driver without doing it in all other drivers. I'm open to adding more fields, but we can't unilaterally do it here. Open a spec ticket to start that discussion for a later release.

https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#api

I noticed that the spec says operation_id is optional, but that appears to be a bug in the spec. A different section of the spec requires it, and you can't correlate bulk write operations that require multiple batches with out it. Happily that doesn't seem to be a problem in the Ruby driver. :-)

https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#bulk-writes

Reply:
OK I will take this out, do you want me to make a PR or ticket for spec to mark operation id required?

Reply:
Send a pull request when you have some free time.